apps/omg_watcher/lib/omg_watcher/exit_processor.ex

```diff
@@ -418,10 +425,11 @@ defmodule OMG.Watcher.ExitProcessor do
     _ = if not Enum.empty?(exits), do: Logger.info("Recognized #{Enum.count(exits)} piggybacks: #{inspect(exits)}")
     {new_state, db_updates} = Core.new_piggybacks(state, exits)
 
-    :ok =
-      exits
-      |> Tools.to_bus_events_data()
-      |> publish_internal_bus_events("InFlightTxOutputPiggybacked")
+    events = Tools.to_bus_events_data(exits)
+    :ok = publish_internal_bus_events(events, "InFlightTxOutputPiggybacked")
+
+    if Code.ensure_loaded?(OMG.WatcherInfo.DB.EthEvent),
+      do: Kernel.apply(OMG.WatcherInfo.DB.EthEvent, :insert_exits!, [events, :in_flight_exit, true])
```

Comment:
all are calls towards `:insert_exits!` idempotent?

Reply:
Yes, exit_processor also persist its state in rocksDB similarly to event listeners.

Reply:
Why do we insert in-flights on piggyback?

If this functions does something more than inserting exits let's call it `insert_exit_events!`.

It also applies to line 530.

Reply:
Important thing here is to spend tx's output that was piggybacked during IFE.
This is done by associate `in_flight_exit` record of EthEvent table with it. It's not piggyback event recorded here.